Source/SVGIconImageCollection.pas

```diff
@@ -0,0 +1,378 @@
+unit SVGIconImageCollection;
+
+interface
+
+{$INCLUDE SVGIconImageList.inc}
+
+uses
+  System.Classes,
+  System.Messaging,
+  SVG,
+  SVGColor,
+  SVGIconItems;
+
+//alias to make usage simpler (less uses clause entries).
+type
+  TSVGIconItem = SVGIconItems.TSVGIconItem;
+  TSVGIconItems = SVGIconItems.TSVGIconItems;
+
+type
+  TSVGIconImageCollection = class;
+
+
+  TSVGCollectionMessage = class(System.Messaging.TMessage)
+    FCollection : TSVGIconImageCollection;
+  public
+    constructor Create(const collection : TSVGIconImageCollection);
+    property Collection : TSVGIconImageCollection read FCollection;
+  end;
+
+  TSVGStopDrawingMessage = class(TSVGCollectionMessage)
+  private
+    FState : boolean;
+  public
+    constructor Create(const collection : TSVGIconImageCollection; const state : boolean);
+    property State : boolean read FState;
+  end;
+
+  TSVGRecreateBitmapsMessage = class(TSVGCollectionMessage);
+
+  TSVGCollectionDestroyedMessage = class(TSVGCollectionMessage);
+
```

Comment:
Cannot destruction be dealt with the standard FreeNotification

Reply:
I guess.. I totally forgot about that, been a while since I wrote any components! will look into it.

Reply:
Done